src/cloudsc_gpu_atlas/CMakeLists.txt

```
@@ -7,119 +7,91 @@
-        ENVIRONMENT "NVCOMPILER_ACC_CUDA_HEAPSIZE=64M"
-    )
-endif()
-
-if( HAVE_CLOUDSC_GPU_SCC_HOIST_ATLAS )
-    ecbuild_add_executable(
-        TARGET dwarf-cloudsc-gpu-scc-hoist-atlas-${prec}
-        SOURCES
-            cloudsc_global_atlas_state_mod.F90
-            ../cloudsc_fortran_atlas/expand_atlas_mod.F90
-            ../cloudsc_fortran_atlas/validate_atlas_mod.F90
-            ../cloudsc_gpu/cloudsc_gpu_scc_hoist_mod.F90
-            cloudsc_driver_scc_mod.F90
-            dwarf_cloudsc_atlas.F90
-        LIBS
-            atlas_f
-            pluto_f
-            cloudsc-common-${prec}-atlas
-        DEFINITIONS ${CLOUDSC_DEFINITIONS} CLOUDSC_GPU_SCC_HOIST
-    )
-
-    ecbuild_add_test(
-        TARGET dwarf-cloudsc-gpu-scc-hoist-atlas
-        COMMAND bin/dwarf-cloudsc-gpu-scc-hoist-atlas
-        ARGS 1 1000 128
-        WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/../../..
-        OMP 1
-    )
-endif()
-
-if( HAVE_CLOUDSC_GPU_SCC_K_CACHING_ATLAS )
-    ecbuild_add_executable(
-        TARGET dwarf-cloudsc-gpu-scc-k-caching-atlas-${prec}
-        SOURCES
-            cloudsc_global_atlas_state_mod.F90
-            ../cloudsc_fortran_atlas/expand_atlas_mod.F90
-            ../cloudsc_fortran_atlas/validate_atlas_mod.F90
-            ../cloudsc_gpu/cloudsc_gpu_scc_k_caching_mod.F90
-            cloudsc_driver_scc_mod.F90
-            dwarf_cloudsc_atlas.F90
-        LIBS
-            atlas_f
-            pluto_f
-            cloudsc-common-${prec}-atlas
-            DEFINITIONS ${CLOUDSC_DEFINITIONS} CLOUDSC_GPU_SCC_K_CACHING
-    )
-
-    ecbuild_add_test(
-        TARGET dwarf-cloudsc-gpu-scc-k-caching-atlas
-        COMMAND bin/dwarf-cloudsc-gpu-scc-k-caching-atlas
-        ARGS 1 1000 128
-        WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/../../..
-        OMP 1
-    )
-endif()
 
+ecbuild_add_executable(
+    TARGET dwarf-cloudsc-gpu-scc-hoist-atlas-${prec}
+    SOURCES
+        cloudsc_global_atlas_state_mod.F90
+        ../cloudsc_fortran_atlas/expand_atlas_mod.F90
+        ../cloudsc_fortran_atlas/validate_atlas_mod.F90
+        ../cloudsc_gpu/cloudsc_gpu_scc_hoist_mod.F90
+        cloudsc_driver_scc_mod.F90
+        dwarf_cloudsc_atlas.F90
+    LIBS
+        atlas_f
+        pluto_f
+        cloudsc-common-${prec}-atlas
+    DEFINITIONS ${CLOUDSC_DEFINITIONS} CLOUDSC_GPU_SCC_HOIST
+)
 
-# Create symlink for the input data
-if( HAVE_SERIALBOX )
-  execute_process(COMMAND ${CMAKE_COMMAND} -E create_symlink
-      ${CMAKE_CURRENT_SOURCE_DIR}/../../data ${CMAKE_CURRENT_BINARY_DIR}/../../../data )
-endif()
+set_target_properties( dwarf-cloudsc-gpu-scc-hoist-atlas-${prec}
+        PROPERTIES
+            Fortran_MODULE_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}/module-dwarf-cloudsc-gpu-scc-hoist-atlas"
+)
 
+ecbuild_add_test(
+    TARGET dwarf-cloudsc-gpu-scc-hoist-atlas-${prec}
+    COMMAND bin/dwarf-cloudsc-gpu-scc-hoist-atlas-${prec}
+    ARGS 1 1000 128
+    WORKING_DIRECTORY ${CMAKE_BINARY_DIR}
+    OMP 1
+)
```

Sorry for not spotting this earlier. Can we add a second test that enables the hoist pool?
```suggestion
ecbuild_add_test(
    TARGET dwarf-cloudsc-gpu-scc-hoist-atlas-${prec}
    COMMAND bin/dwarf-cloudsc-gpu-scc-hoist-atlas-${prec}
    ARGS 1 1000 128
    WORKING_DIRECTORY ${CMAKE_BINARY_DIR}
    OMP 1
)
ecbuild_add_test(
    TARGET dwarf-cloudsc-gpu-scc-hoist-pool-atlas-${prec}
    COMMAND bin/dwarf-cloudsc-gpu-scc-hoist-atlas-${prec}
    ARGS 1 1000 128
    WORKING_DIRECTORY ${CMAKE_BINARY_DIR}
    OMP 1
    ENVIRONMENT HOIST_POOL=1
)
```

---

Thanks for the hint. It has been added.